packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -613,40 +631,104 @@ export class TSRHandler {
 			}
 		}
 
-		await Promise.race([
-			Promise.all(ps),
-			new Promise<void>((resolve) =>
+		const resultsOrTimeout = await Promise.race([
+			Promise.all(ps).then(
+				(result) =>
+					({
+						success: true,
+						result,
+					} as UpdateDeviceOperationsResult)
```

Comment:
Casting as `UpdateDeviceOperationsResult` here doesn't provide type safety (one example which goes unnoticed is `{}`)

Use `literal<UpdateDeviceOperationsResult>()`, or `await Promise.race<UpdateDeviceOperationsResult>` instead.